### PR TITLE
feat: update workspace util func and fix known bug

### PR DESCRIPTION
**Pull request by healthjyk**
…ce dir is not hidden folder

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

1. add CURD func by default workspace operator;
2. use inline for the `GenericObjectStorageConfig` for `OssConfig` and `S3Config`;
3. correct validation for `S3Config`.

### CODE REVIEW

**Comment by SparkYuan** [comment]
What is the `pkg/workspace/testdata/invalid_workspaces_dir/dir_workspace_file/.gitkeep` file used for?

File: pkg/workspace/operator.go [hunk 1]
```diff
@@ -28,6 +28,60 @@ var (
...
+	operator, err := NewValidDefaultOperator()
+	if err != nil {
+		return err
+	}
+	return operator.DeleteWorkspace(name)
+}
+
 // Operator is used to handle the CURD operations of workspace. Operator only supports local file
 // system as backend for now.
 type Operator struct {
```
Comment (SparkYuan) [inline pkg/workspace/operator.go:87]: Why not define the Operator as an interface as we will have various types of operators in the future?

Comment (healthjyk) [reply pkg/workspace/operator.go:87]: That's a good suggestion! But it's for the future, now it's more visualized.
Now it seems C/U/R/D defines the functions of the interface "Operator", but if there is any more functions need, we can get more feedback from may-exist users and subsequent R&D work.
Besides, even though the operator becomes an interface, there still needs a struct holding the structure of the configuration of the Operator, which may have two field `backend type` and `backend config`, and it's a struct.
Defining Operator an interface is a good idea, but I think we can do it when there are more workspace operator backends, now the struct is ok. What's your opinion?

**Review by SparkYuan — APPROVED** [review]
lgtm

**Pull request merged**